### PR TITLE
feat: Setup additional node tags

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Imports new tags from planx-core (will bump across all projects once https://github.com/theopensystemslab/planx-core/pull/528 merged to `main`)
 - Tags are coloured as per https://opensystemslab.slack.com/archives/C04DZ1NBUMR/p1728385666798109 (OSL Slack)
 - Adds toggle to view tags, peristed via local storage

<p align="center">

https://github.com/user-attachments/assets/d48e46ef-1cd6-49ae-9d8f-a792b4742dd3

Screenshot 2024-10-08 at 14 14 42

</p>

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/components/FlowEditor/ToggleTagsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import LabelOffIcon from "@mui/icons-material/LabelOff";
+import Box from "@mui/material/Box";
+import IconButton from "@mui/material/IconButton";
+import { styled } from "@mui/material/styles";
+import Tooltip, { tooltipClasses, TooltipProps } from "@mui/material/Tooltip";
+import { useStore } from "pages/FlowEditor/lib/store";
+import React from "react";
+import { FONT_WEIGHT_SEMI_BOLD } from "theme";
+
+const TooltipWrap = styled(({ className, ...props }: TooltipProps) => (
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/FlowEditor/ToggleTagsButton.tsx:11]: This is repeated a few places now, I'll open up a follow up PR to make a variant or standardise this via the theme.

File: editor.planx.uk/src/pages/FlowEditor/components/FlowEditor/ToggleTagsButton.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          aria-label="Toggle tags"
+          onClick={toggleShowTags}
+          size="large"
+          sx={(theme) => ({
+            color: showTags
+              ? theme.palette.text.primary
+              : theme.palette.text.disabled,
+          })}
+        >
+          {showTags ? <LabelIcon /> : <LabelOffIcon />}
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/FlowEditor/ToggleTagsButton.tsx:52]: This pattern _may_ be restricting us more than we need to by limiting us to using "off" MUI icons if we want to expand this to toggle images or data fields?

https://mui.com/material-ui/material-icons/?query=off

File: editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx [hunk 3]
```diff
@@ -1,35 +1,60 @@
 import Box from "@mui/material/Box";
-import { visuallyHidden } from "@mui/utils";
+import { PaletteOptions, useTheme } from "@mui/material/styles";
 import { NodeTag } from "@opensystemslab/planx-core/types";
+import { useStore } from "pages/FlowEditor/lib/store";
 import React from "react";
+import { getContrastTextColor } from "styleUtils";
 import { FONT_WEIGHT_SEMI_BOLD } from "theme";
 
 export const TAG_DISPLAY_VALUES: Record<
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx:9]: Not sure if I have the right colour scheme or setup here @ianjon3s , should be be a quick and easy fix though!

File: editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx [hunk 4]
```diff
@@ -1,35 +1,60 @@
...
-    {TAG_DISPLAY_VALUES[tag].displayName}
-  </Box>
-);
+export const Tag: React.FC<{ tag: NodeTag }> = ({ tag }) => {
+  const theme = useTheme();
+
+  const showTags = useStore((state) => state.showTags);
+  if (!showTags) return null;
+
+  const tagBgColor = theme.palette.nodeTag[TAG_DISPLAY_VALUES[tag].color];
```
Comment (jamdelion) [inline editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx:41]: Glad you've extracted this out but it's still quite horrible how nested it is 😂

File: editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx [hunk 5]
```diff
@@ -1,35 +1,60 @@
...
+import { getContrastTextColor } from "styleUtils";
 import { FONT_WEIGHT_SEMI_BOLD } from "theme";
 
 export const TAG_DISPLAY_VALUES: Record<
   NodeTag,
-  { color: string; displayName: string }
+  { color: keyof Palette["nodeTag"]; displayName: string }
 > = {
   placeholder: {
-    color: "#FAE1B7",
```
Comment (jamdelion) [inline editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Tag.tsx:12]: Very much approve of the removal of colour hex codes 😁

**Review by jamdelion — APPROVED** [review]

**Pull request merged**